### DISCUSSION TITLE
API Fetch/Sorting - Rendering takes a while and crashes eventually

### DISCUSSION THREAD

**Discussion by mzschunke** [neuefische/questions#595] [Web] [answered]
Hi all,

I want to fetch player data from this API: https://www.balldontlie.io/api/v1/players and sort them alphabetically to render a simple list, starting with the letter 'a'. Unfortunately, the data is not presorted, hence why I implemented the filter method for each requested page. It works most of the times (takes a while to render) - however I sometimes get an error message and a "429 / too many requests" response. 

I am thankful for any ideas how to deal with this issue! 

Here is the code: 

```
import useSWR from "swr";
import { useState, useEffect  } from "react";

export default function PlayerOverview() {
  const URL = "https://www.balldontlie.io/api/v1/players";
  const PAGE_SIZE = 100;
  const TOTAL_PAGES = 52; 
  const [page, setPage] = useState(1);
  const { data, error, isLoading } = useSWR(`${URL}?per_page=${PAGE_SIZE}&page=${page}`);
  const [players, setPlayers] = useState([]);

  useEffect(() => {
    if (data && data.data) {
      const fetchedPlayers = data.data;
      const filteredPlayers = fetchedPlayers.filter((player) =>
        player.last_name.toLowerCase().startsWith("a")
      );
      setPlayers((prevPlayers) => [...prevPlayers, ...filteredPlayers]);
      console.log(players);
      if (page < TOTAL_PAGES) {
        setPage((prevPage) => prevPage + 1);
      }
    }
  }, [data]);

if (error) return <div>Failed to load</div>;
if (isLoading) return <div>Loading....</div>;

return (
  <div>
    <h1>Players</h1>
    <ul>
      {players.map((player) => (
        <li key={player.id}>
          {player.last_name}{", "}{player.first_name}
        </li>
      ))}
    </ul>
  </div>
);
}
```

**Comment by actyralf** [comment, marked as answer]
I guess there is a usage limit on this API - you have to check the terms & conditions.
One thing that might help: useSWR has a ["revalidate on focus"](https://swr.vercel.app/docs/revalidation) option that causes a new fetch whenever the related page gets the focus. You could try to switch this off. Also, try to [inspect your network traffic](https://developer.chrome.com/docs/devtools/network/) to investigate unnecessary requests.
